lms/db/_locks.py

```diff
@@ -0,0 +1,41 @@
+from enum import IntEnum
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+
+
+class CouldNotAcquireLock(Exception):
+    """Exception raised if a lock cannot be immediately acquired."""
```

Comment:
```suggestion
    """A lock couldn't be acquired because another process was already holding it."""
```

I like to name exceptions and their docstrings after the error condition they represent, rather than describing when they're raised. It leads to more concise docstrings and you can pack more info in. For example:

```python
class HTTPRequestTimedOut(Exception):
    """An HTTP request failed because it timed out."""
```

Rather than:

```python
class HTTPRequestTimedOut(Exception):
    """Exception raised when an HTTP request times out."""
```

The fact that the class is an exception class and is raised when the described situation occurs is kind of a truism, so starting every exception docstring with "Exception raised when..." is kind of a waste of space.